internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -437,6 +477,12 @@ func (r *ClusterDeploymentsReconciler) updateIfNeeded(ctx context.Context,
 		update = true
 		notifyInfraEnv = true
 	}
+	// update hyperthreading settings
+	if hyperthreadingInSpec(clusterInstall) {
```

Comment:
so we are always updating the cluster in case we have control plane nodes or compute node? don't we want to compare with the values in the cluster?

Reply:
Do we do this with other configurations? I noticed that we take the Spec and configure as-is  but does Hyperthreading have special requirements that mandate the comparison?

Reply:
Implemented the state check prior to update to avoid unnecessary configurations